src/common.rs

```diff
@@ -3,11 +3,35 @@ use std::{collections::BTreeMap, fmt};
 use combine::easy::{Error, Info};
 use combine::{choice, many, many1, optional, position, StdParseResult};
 use combine::{parser, Parser};
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
 
 use crate::helpers::{ident, kind, name, punct};
 use crate::position::Pos;
 use crate::tokenizer::{Kind as T, Token, TokenStream};
 
+#[cfg(feature = "serde")]
```

Comment:
Not sure if this is a _good_ idea, but I think to not duplicate everything you could do something like:

```rust
#[cfg(feature = "serde")]
mod text_traits {
    use serde::{Serialize, Deserialize};

    pub trait SerdeTraits<'a>: Serialize + Deserialize<'a> {}

    impl<'a, T> SerdeTraits<'a> for T where T: Serialize + Deserialize<'a> {}
}

#[cfg(not(feature = "serde"))]
mod text_traits {
    pub trait SerdeTraits<'a> {}

    impl<'a, T> SerdeTraits<'a> for T {}
}

// Define the main trait with conditional compilation
pub trait Text<'a>: 'a {
    type Value: 'a
        + From<&'a str>
        + AsRef<str>
        + std::borrow::Borrow<str>
        + PartialEq
        + Eq
        + PartialOrd
        + Ord
        + fmt::Debug
        + Clone
        + text_traits::SerdeTraits<'a>;
}
```